x-pack/plugins/cloud_security_posture/server/lib/fleet_util.ts

```diff
@@ -103,13 +90,37 @@ export const getCspPackagePolicies = (
 ): Promise<ListResult<PackagePolicy>> => {
   const sortField = queryParams.sort_field?.replaceAll(BENCHMARK_PACKAGE_POLICY_PREFIX, '');
 
-  return packagePolicyService.list(soClient, {
-    kuery: getPackageNameQuery(postureType, packageName, queryParams.benchmark_name),
-    page: queryParams.page,
-    perPage: queryParams.per_page,
+  const allCSPPackages = await packagePolicyService.list(soClient, {
+    kuery: getPackageNameQuery(),
+    page: 1,
+    perPage: 10000,
     sortField,
     sortOrder: queryParams.sort_order,
   });
+
+  const filteredItems = allCSPPackages.items.filter(
+    (pkg) =>
+      pkg.inputs.filter((input) =>
+        postureType === 'all'
+          ? input.enabled
+          : input.enabled && input.policy_template === postureType
```

Comment:
_nit_ you can reduce code repetition by:
`input.enabled && (postureType === 'all' || input.policy_template === postureType)`